### PR TITLE
Create a Code of conduct to complete the Github community checklist

### PR DESCRIPTION
**Pull request by dheerajbhaskar**
This can be a working Code of Conduct. This is based on http://contributor-covenant.org which is generally regarded as suitable for Open Source projects

## Purpose

Create a Code of conduct to complete the Github community checklist (see image below)
Github Community Checklist

## Proposed changes

- Add a working code of conduct which is generally used in open source projects (one of github's templates)

## TODO

- [x] Find a relevant template for code of conduct
- [x] Find a relevant email address of reactos org for reporting issues (info@reactos.org)

### CODE REVIEW

**Comment by dheerajbhaskar** [comment]
Please let me know if any changes are needed

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment include:
+
+* Using welcoming and inclusive language
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Sometimes I might not want to be inclusive and thus don't use inclusive language. I could argue that demanding inclusive language discriminates against people, based on their language.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: This would discriminate against people with autism, EDD, or Asperger's Syndome.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Sounds like "Do not claim that the code is shitty or wrong, it is just written from a differing viewpoint with different experience."

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+* The use of sexualized language or imagery and unwelcome sexual attention or advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a professional setting
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Who are "Project Maintainers"? All core developers? They surely don't have this responsibility. Where does this responsibility come from? Why would we have that? This is a claim that we cannot fulfil. We are developers, not managers / HR guys.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Again: Remove responsibility.

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at info@reactos.org. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Do we even have an info@reactos.org?

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership.
```
Comment (tkreuzer) [inline CODE_OF_CONDUCT.md]: Again this demands responsibilites ("enforce") that we cannot and do not want to have.

**Comment by tkreuzer** [comment]
IMO this this thing is way over the top. It makes unneccessary and unsustainable claims about our responsibilities towards outside people. We are not responsible for the behaviour of other people. A simple "Don't be an asshole, or we will kick you out." should be enough. ReactOS is not a safe space for people, who can't tolerate criticism and not a playground for people without any programming experience.

**Comment by mifritscher** [comment]
Yes, this one-liner should be enough. If we get more management some day it can be extend anyway.

**Comment by dheerajbhaskar** [comment]
@tkreuzer Good to go? :)

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (dheerajbhaskar) [reply CODE_OF_CONDUCT.md]: Well I guess so, that's what is listed on the website (see screenshot below)

Email listing screenshot

**Comment by blackcrack** [comment]
:+1:  Now the whole thing has a certain quality and looks really good, well done ! 
well done.

**Comment by HBelusca** [comment]
ROTFL xD

File: CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Code of Conduct
+
+Don't be an asshole, or we will kick you out.
```
Comment (gigaherz) [inline CODE_OF_CONDUCT.md]: I would probably prefer this to say something closer to:

> Be respectful toward other members of the community. Don't initiate or engage in discussions that are clearly designed to insult, inflame, attack, or incite hate/discrimination/bullying against others. Don't feed the trolls.
> 
> In other words: Don't be an asshole.
>
> If we detect any kind of behaviour we consider assholey, we reserve the right to ban, block, or remove the involved members from any or all of our communities.

**Comment by dheerajbhaskar** [comment]
@gigaherz
Depends on what the maintainers want. Looks like they want to keep it simple.

I'll wait for them to merge this PR or suggest a change

**Comment by binarymaster** [comment]
I like @gigaherz's version, it sounds better and looks more serious than just one liner.

**Comment by dheerajbhaskar** [comment]
@binarymaster shall I update it to gigaherz's version?

**Comment by gigaherz** [comment]
I just gave a suggestion. Also I am a maintainer, @binarymaster is just a valued contributor. :P

**Comment by blackcrack** [comment]
This code of conduct should be professional and not like kindergarten.
This should encourage animate and have a certain respect and have not stated that reactos consists of a kindergaten, who think it must always make a joke and others refer, best of all, the sandburg of others short times to throw down.
I think companies and respectful individuals should be able to do this without being addressed like a child, that can be very frightening and dismissive.
Here are not working only two big countries working together, but the whole world!
and in this HTP(hackers)-sence should Reactos a really pretty well and respectful Code of Conduct have to it be in able that even Bill Gates or Kevin David Mitnick there working with and have not a bad feel there..
You are here responsible for putting up a code that allows every great programmer to get involved without the code of conduct being offensive !

Think of something ingenious, certainly something of the past would also be helpful to bring the leadership over .. for example, in the spirit of co-working as in former times "Hack the Planet" in the sense of cooperation and self-employment in the community and of the community, as in thinking of Linux but this time a self - takeover and self-buildup of a WinNT platform to allow anyone to run their WinNT programs for free without paying for horrendous rumbling  sum and that every code line that creates a child or another soul also has the possibility to build up Computer with a free and open source without being liable to prosecution and thus gives the possibility to have a free and open code computer without putting down a cent by peoples who can not afford it . An operating system should be something like a fundamental right in these times.

Some thing and not a chop off anything that fits.
in this direction could we already take the windowscode from Microsoft and tell to be made by us !
This Code of Conduct should also made by Reactos .

best regards
Blacky

**Comment by ThFabba** [comment]
@blackcrack TL;DR. Please try to make your point without a wall of text.

**Comment by dheerajbhaskar** [comment]
@gigaherz Whoops! I'm sorry, the contributor and member tags threw me off. I for some reason thought a contributor is a maintainer :sweat_smile: 
Member tag

Do let me know if you want me to update the code to your version, I'd be happy to do it :slightly_smiling_face: 

---

@sanchaez I used the Contributor Covenant because Github mentioned it's for projects of all sizes while Citizen Code of Conduct was primarily for larger projects. 
conduct templates

I'm not sure what changes to make based on your suggestion. Would you suggest that we use the Citizen Code of Conduct? @tkreuzer suggested we have a simple one. 

If you and the other Members can decide what the code of conduct should be, I'll be happy to push in the desired changes :slightly_smiling_face: 

---

@blackcrack To summarize his points:
1. He'd want a respectful code of conduct, not one which treats developers like children.
2. He wouldn't want the code of conduct to be offensive.

He goes on about inspiration, but I think above are two main points he makes. Blacky, it'd be great if you make a concrete proposal for a code of conduct rather than a principle or a guideline for it. It'll help us finalize things faster.

PS: If you did use google translate, it'd be great if you can post both the original and translated texts

**Comment by dheerajbhaskar** [comment]
@sanchaez reconsider using the Citizen Code of Conduct. It might not be what you're looking for. It has a lot of requirements (listed below). You as members might not have or be even inclined to have that much time devoted to administration at this point.

Requirements of the Citizen Code of Conduct:
requirements for citizen code of conduct

**Comment by sanchaez** [comment]
There are more fields but the actual text doesn't contain _responsibility_ but _possibility_ instead, meaning there is less administering burden.

21 мар. 2018 г. 13:46 пользователь "Dheeraj Bhaskar" <notifications@github.com<mailto:notifications@github.com>> написал:

@sanchaez<https://github.com/sanchaez> reconsider using the Citizen Code of Conduct. It might not be what you're looking for. It has a lot of requirements (listed below). You as members might not have or be even inclined to have that much time devoted to administration at this point.

Requirements of the Citizen Code of Conduct:
[requirements for citizen code of conduct]<https://user-images.githubusercontent.com/2944909/37708104-734d5c50-2d2b-11e8-91ab-71b47496cbc3.PNG>

**Comment by blackcrack** [comment]
@dheerajbhaskar
i use it already (google translater(qtranslate)) ;)
(this make's me a bit disappointed, because I'm trying already and try my bestststs *s*)

**Comment by sanchaez** [comment]
@dheerajbhaskar I think we can raise CoC question in our next meeting and notify you about the results.

**Comment by blackcrack** [comment]
@sanchaez 
in the next Meeting should you also speak about reactos/rapps-db submits,
this should connected with Forums/Reactos Login to be able for submit new 
files or program updates into the app-database,
because Applications texts be not a programming code but a support 
for reactos and should be able for Forumsmembers enable to do that. 
Or anybody else speak this on, because realnames for the Appdb in Rrapps
is exaggerated and any who it is registered in the Reactos Forum 
should be in able to update the Applications.

best regards
Blacky

**Comment by gigaherz** [comment]
@blackcrack We have every intention on having a web-based compatibility frontend, in the future. The rapps-db repository is a necessary temporary method, until we get a new web maintainer, who can finish the "application compatibility and binary submission" platform.

**Comment by SHwareSystemsDev** [comment]
If a formal Code is adopted or not, the ReadMe.md here (and most pages on the reactos.org website) should provide a reference to a static reactos.org/about page with the info from the wiki Foundation page on it. Whether conduct breach enforcement is considered a responsibility or possibility, ultimately that is who is accountable now for any actions. Having to hunt that info down in the bowels of the Wiki does more to detract from people considering this a legitimate project they want to be a part of than any occasional abusive remarks or other breaches those Codes cover, imho. The site looks more like a phishing attempt without links like that, to info that is easily verifiable through other means as being legitimate.

**Comment by blackcrack** [comment]
@gigaherz 
Hi David,
thank you for the information :)
> That said, even if we do have that site, there is a chance it may require real names associated with those submissions, for matters of liability.

well, not any want give in the world his/her realname it should be possible to use this with the Forum Account and should be possible for any who have a login to make with and give a helping hand.
Realname disappoint securely not me alone. it has works in the past to have possible to working together without Realnames, it should works today also by something like Program updates and Program adding
to be able for this. I can see that you need at the program is mandatory for proof, who and when a code is written, but by Program adding on Rapps or maintaining /updating, is this imho not realy need, only a login as user in the forum or maybe Login in the Forum and in the Group of Rapps-Maintainer without Realname, this is maybe a possible. To make it possible for peoples for help there if they are taking in
a Rapps group for testing if they are reliable , like in the past.. from Real names i do not think much, just so much that you can put the name down to profile yourself... and later profit from it in other way..
I do it, because i whant it and have fun, i don't need set (or sowed) my name, i want have fun to do this,
more not.. and with this can i help others to have the actual versions and at last me to, to have a actual version.. like the bashscripts where i have under blackysgate.de/files who i make  or in store.kde.org the Iconthemes, it make me fun and i guess, there gives other peoples who see it same.. Updating and maintaining the programs in rapps should be able for Forums members without real names.

best regards
Blacky

**Comment by ThFabba** [comment]
Reminder: this is not a discussion forum about all the things you want to improve about ROS.
It is a concept review about a code of conduct. Suggestions about how to get the proposed commits into a state that is agreeable to the majority and can be accepted into master belong here. Nothing else.

Please stop posting off-topic comments or I will start deleting them.

**Comment by JohnBlood** [comment]
I am not a fan of Codes of Conduct because they can become political weapons. http://podcast.asknoahshow.com/12

**Comment by gigaherz** [comment]
After some internal discussion, we have come up with this version:

Be respectful toward other members of the community. Don't initiate or engage in discussions that are designed to insult, inflame, attack, or incite hate/discrimination/bullying against others. Don't feed the trolls. If we detect any kind of behaviour we consider unacceptable, we reserve the right to ban, block, or remove the involved members from any or all of our communities.

**Comment by sanchaez** [comment]
I'm still not fond of such implicit documents, but I won't oppose either.

**Comment by dheerajbhaskar** [comment]
@gigaherz thanks for the update. I've revised the CoC to have the text you've shared.

**Comment by tkreuzer** [comment]
@sanchaez I don't like these comprehensive and explicit versions, because IMO it suggests that our community is full of children, who need to be parented and sheltered by us and that we want these people in our community, rather than grown-ups, who can estimate themselves what behaviour is acceptable. I also don't like to be part of a universal "safe-space" where one has to weigh every word, and compare it bulletpoint by bulletpoint against our "speech laws".

**Pull request merged**